examples/pytorch/schnet/layers.py

```diff
@@ -0,0 +1,290 @@
+        self.linear_layer2 = nn.Linear(self._dim, self._dim)
+
+        if act == "sp":
+            self.activation = nn.Softplus(beta=0.5, threshold=14)
+        else:
+            self.activation = act
+
+    def update_edge(self, edges):
+        rbf = edges.data["rbf"]
+        h = self.linear_layer1(rbf)
+        h = self.activation(h)
+        h = self.linear_layer2(h)
+        return {"h": h}
+
+    def forward(self, g):
+        g.apply_edges(self.update_edge)
+        g.update_all(message_func=fn.u_mul_e('new_node', 'h', 'neighbor_info'),
+                     reduce_func=fn.sum('neighbor_info', 'new_node'))
+        return g.ndata["new_node"]
+
+
+class Interaction(nn.Module):
+    """
+    The interaction layer in the SchNet model.
+    """
+
+    def __init__(self, rbf_dim, dim):
+        super().__init__()
+        self._node_dim = dim
+        self.activation = nn.Softplus(beta=0.5, threshold=14)
+        self.node_layer1 = nn.Linear(dim, dim, bias=False)
+        self.cfconv = CFConv(rbf_dim, dim, act=self.activation)
+        self.node_layer2 = nn.Linear(dim, dim)
+        self.node_layer3 = nn.Linear(dim, dim)
+
+    def forward(self, g):
+
+        g.ndata["new_node"] = self.node_layer1(g.ndata["node"])
+        cf_node = self.cfconv(g)
+        cf_node_1 = self.node_layer2(cf_node)
+        cf_node_1a = self.activation(cf_node_1)
+        new_node = self.node_layer3(cf_node_1a)
+        g.ndata["node"] = g.ndata["node"] + new_node
+        return g.ndata["node"]
+
+
+class VEConv(nn.Module):
+    """
+    The Vertex-Edge convolution layer in MGCN which take edge & vertex features
+    in consideratoin at the same time.
+    """
+
+    def __init__(self, rbf_dim, dim=64, update_edge=True):
+        """
+        Args:
+            rbf_dim: the dimension of the RBF layer
+            dim: the dimension of linear layers
+            update_edge: whether update the edge emebedding in each conv-layer
+        """
+        super().__init__()
+        self._rbf_dim = rbf_dim
+        self._dim = dim
+        self._update_edge = update_edge
+
+        self.linear_layer1 = nn.Linear(self._rbf_dim, self._dim)
+        self.linear_layer2 = nn.Linear(self._dim, self._dim)
+        self.linear_layer3 = nn.Linear(self._dim, self._dim)
+
+        self.activation = nn.Softplus(beta=0.5, threshold=14)
+
+    def update_rbf(self, edges):
+        rbf = edges.data["rbf"]
+        h = self.linear_layer1(rbf)
+        h = self.activation(h)
+        h = self.linear_layer2(h)
+        return {"h": h}
+
+    def update_edge(self, edges):
+        edge_f = edges.data["edge_f"]
+        h = self.linear_layer3(edge_f)
+        return {"edge_f": h}
+
+    def forward(self, g):
+        g.apply_edges(self.update_rbf)
+        if self._update_edge:
+            g.apply_edges(self.update_edge)
+
+        g.update_all(message_func=[
+            fn.u_mul_e("new_node", "h", "m_0"),
+            fn.copy_e("edge_f", "m_1")
+        ],
```

Comment:
Could you please improve the indent format here?

Reply:
😂 The autopep8 seems cannot handle this. 
Maybe this one will be better?
```py
       g.update_all(message_func=[
           fn.u_mul_e("new_node", "h", "m_0"),
           fn.copy_e("edge_f", "m_1")],
                    reduce_func=[
           fn.sum("m_0", "new_node_0"),
           fn.sum("m_1", "new_node_1")])
```

Reply:
```suggestion
g.update_all(
    message_func=[
        fn.u_mul_e("new_node", "h", "m_0"),
        fn.copy_e("edge_f", "m_1")],
    reduce_func=[
        fn.sum("m_0", "new_node_0"),
        fn.sum("m_1", "new_node_1")])
```
How about this?

Reply:
Great~